cuML/src/solver/cd.h

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2018-2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "ml_utils.h"
+#include <cuda_utils.h>
+#include <linalg/gemm.h>
+#include <linalg/add.h>
+#include <linalg/subtract.h>
+#include <linalg/multiply.h>
+#include <linalg/eltwise.h>
+#include <linalg/unary_op.h>
+#include <linalg/cublas_wrappers.h>
+#include <matrix/math.h>
+#include <matrix/matrix.h>
+#include "glm/preprocess.h"
+#include "shuffle.h"
+#include <functions/penalty.h>
+#include <functions/softThres.h>
+#include <functions/linearReg.h>
+
+namespace ML {
+namespace Solver {
+
+using namespace MLCommon;
+
+template<typename math_t>
+void cdFit(math_t *input,
```

Comment:
Like other algos, I believe we'd want to expose `cumlHandle_impl` here. @jirikraus has provided some very good set of guidelines for doing so in multiple PRs: #481 , #482 and #394 .

Reply:
Will update all the algorithms together for 0.8 release.